README.md

```diff
@@ -0,0 +1,617 @@
+  Don't forget to visit our website {{ asyncapi.externalDocs().url() | urlize | safe }}.
+{% endif %}
+```
+
+#### What Else I Can Do?
+
+You can for example have conditions like in [partials/diagramContent.html](partials/diagramContent.html) file:
+```html
+{% if asyncapi.hasComponents() and asyncapi.components().schemas() %}
+...
+{% else %}
+...
+{% endif %}
+```
+
+Nunjucks templating engine is very powerful with many different options. Whenever you develop a template, always consult not only with Generator docs but also [Nunjucks docs](https://mozilla.github.io/nunjucks/templating.html).
+
+### Partials
+
+In the `partials` directory, you can find two files, one is an example of Nunjucks [include](https://mozilla.github.io/nunjucks/templating.html#include), and another one is a [macro](https://mozilla.github.io/nunjucks/templating.html#macro). Use this feature to partition your Template into smaller chunks that make future maintenance of the Template easier.
+
+#### Includes
+
+Check our [partials/diagramContent.html](partials/diagramContent.html) file to see how `includes` work. It is just a simple file with content that can be included in another file. Have a look at [template/index.html](template/index.html) file to understand how to use include:
+```html
+{% include "partials/diagramContent.html" %}
+```
+
+#### Macros
+
+Macros are like dynamic includes that change depending on the input that you pass to the macro. It is like a function with arguments. Check out [partials/listChannels.html](partials/listChannels.html) file to see an example macro called `listChannels`:
+
+```html
+{% macro listChannels(channels, operationType) %}
+<h2>
+  Channels that you can {{ operationType }} to
+</h2>
+<hr />
+<br />
+<div class="container mx-auto px-8">
+  <ul class="list-disc"></ul>
+    {% for channelName, channel in channels %}
+      {% if operationType === 'publish' %} 
+          {% if channel.hasPublish() %} 
+              <li><strong>{{ channelName }}</strong></li>
+          {% endif %}
+      {% elif operationType === 'subscribe' %}
+          {% if channel.hasSubscribe() %} 
+              <li><strong>{{ channelName }}</strong></li>
+          {% endif %}
+      {% endif %}
+    {% endfor %}
+  </ul>
+</div>
+{% endmacro %}
+```
+
+This macro is used in the [template/index.html](template/index.html) file:
+```
+{% from "partials/listChannels.html" import listChannels %}
+...
+{{ listChannels(asyncapi.channels(), 'subscribe') }}
+{{ listChannels(asyncapi.channels(), 'publish') }}
+```
+
+### File Templates
+
+The Generator has a feature called `file templates` that allows you to create a template file with special that has `$$` markers, like `$$schema$$`. There are multiple different file templates available. In this Template, during generation, `$$schema$$` is replaced with a schema name, and Template gets the following variables in the context:
```

Comment:
Maybe a link to https://github.com/asyncapi/generator/blob/master/docs/authoring.md#file-templates?